### PR TITLE
report_table(): let's make this right

### PR DESCRIPTION
**Pull request by strengejacke**
#289

### CODE REVIEW

**Comment by strengejacke** [comment]
@DominiqueMakowski I remember times where I _always_ commited to master and never used branches/PRs, and you were exactly the opposite... ;-)

**Comment by DominiqueMakowski** [comment]
You are a bad influence 😈

**Comment by strengejacke** [comment]
I think a `format.report_table()` method is missing. But why did you add `display.report_table()` to _parameters_? Isn't it better to have it all here in one place?

**Comment by DominiqueMakowski** [comment]
> But why did you add display.report_table() to parameters? Isn't it better to have it all here in one place?

I thought so too, but for some reason it kept saying that `display.parameters_model is not an object from parameters`...

**Comment by DominiqueMakowski** [comment]
On hindsight, I probably did something wrong though, feel free to fix

**Comment by strengejacke** [comment]
I see, strange...

**Comment by strengejacke** [comment]
``` r
library(report)

results1 <- t.test(mpg ~ am, data = mtcars)
results2 <- t.test(wt ~ am, data = mtcars)
results3 <- t.test(qsec ~ am, data = mtcars)

c(
  report_table(results1),
  report_table(results2),
  report_table(results3)
) |>
  display()
#> Warning: Unable to retrieve data from htest object. Using t_to_d()
#> approximation.

#> Warning: Unable to retrieve data from htest object. Using t_to_d()
#> approximation.

#> Warning: Unable to retrieve data from htest object. Using t_to_d()
#> approximation.
```

| Parameter | Group | Mean_Group1 | Mean_Group2 | Difference |     95% CI      |   t   |  df   |    p    |   d   |      d CI      |
|:----------|:-----:|:-----------:|:-----------:|:----------:|:---------------:|:-----:|:-----:|:-------:|:-----:|:--------------:|
| mpg       |  am   |    17.15    |    24.39    |   -7.24    | (-11.28, -3.21) | -3.77 | 18.33 |  0.001  | -1.76 | (-2.82, -0.67) |
| wt        |  am   |    3.77     |    2.41     |    1.36    |  (0.85, 1.86)   | 5.49  | 29.23 | \< .001 | 2.03  |  (1.13, 2.91)  |
| qsec      |  am   |    18.18    |    17.36    |    0.82    |  (-0.49, 2.14)  | 1.29  | 25.53 |  0.209  | 0.51  | (-0.28, 1.29)  |

Fixed Effects

<sup>Created on 2022-10-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

File: R/report_table.R [hunk 1]
```diff
@@ -132,19 +133,29 @@ print.report_table <- function(x, ...) {
...
 }
 
 
+#' @export
+display.report_table <- function(object, ...) {
+  # fix caption
+  if (is.null(list(...)$caption)) {
+    attr(object, "no_caption") <- TRUE
+  }
+  class(object) <- c("report_table", "parameters_model", "data.frame")
```
Comment (DominiqueMakowski) [inline R/report_table.R:154]: why not 
`class(object) <- c("parameters_model, class(object))`

Comment (strengejacke) [reply R/report_table.R:154]: I think then `format.report_table()` is ignored, which is not what we want?

Comment (DominiqueMakowski) [reply R/report_table.R:154]: right

**Review by DominiqueMakowski — APPROVED** [review]

**Pull request merged**